tests/Makefile.am

```diff
@@ -180,13 +168,5 @@ yamitranscode_SOURCES  = vppinputdecode.cpp vppinputoutput.cpp vppoutputencode.c
 
 bin_PROGRAMS += yamiinfo
 yamiinfo_SOURCES = yamiinfo.cpp
-yamiinfo_LDADD = \
-	$(top_builddir)/decoder/libyami_decoder.la \
-	$(top_builddir)/encoder/libyami_encoder.la \
-	$(top_builddir)/vpp/libyami_vpp.la \
-	$(NULL)
-yamiinfo_LDFLAGS = -Wl,--no-as-needed \
-	$(YAMI_DECODE_LDFLAGS) \
-	$(YAMI_ENCODE_LDFLAGS) \
-	$(YAMI_VPP_LDFLAGS) \
-	$(NULL)
+yamiinfo_LDADD = $(YAMI_COMMON_LIBS)
+yamiinfo_LDFLAGS = $(YAMI_DECODE_LDFLAGS)
```

Comment:
Ok, I just checked on Ubuntu and we still need `-Wl,--no-as-needed` for yamiinfo, unconditionally.